src/web3-utils.js

```diff
@@ -218,6 +218,14 @@ export async function getMainAccount(web3) {
   }
 }
 
+export async function getLatestBlockTimestamp(web3) {
+  const latestBlockInfo = await web3.eth.getBlock('latest')
+  if (!latestBlockInfo) {
+    return 0 // we should have a retry mechanism
+  }
+  return latestBlockInfo.timestamp
```

Comment:
I'm still a bit wary of returning the raw timestamp here; I know quite a few people who have been burned by the difference in units (seconds vs. milliseconds).

I'd still suggest returning a Date object or at least ms unix time.